Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -222,9 +241,24 @@ protected override void ProcessRecord()
 
             // Content type to use for the new object.
             string objContentType = null;
-
             Stream contentStream = null;
-            if (!string.IsNullOrEmpty(File))
+
+            if (ParameterSetName == ParameterSetNames.ContentsFromFolder)
+            {
+                // user gives us the path to a folder, we will resolve the path and upload the content of that folder
+                string resolvedFolderPath = GetFullPath(Folder);
+                if (string.IsNullOrWhiteSpace(resolvedFolderPath) || !Directory.Exists(resolvedFolderPath))
+                {
+                    throw new DirectoryNotFoundException(string.Format("Directory {0} cannot be found", resolvedFolderPath));
```

Comment:
Please use the string interpolation operator. Also, putting quotation marks around the directory name makes it easier to diagnose errors.

````
$"Directory '{resolvedFolderPath}' be found."
````